crypto/tls/tls.go

```diff
@@ -79,10 +90,15 @@ func (cfg *ClientConfig) GetTLSConfig() (*tls.Config, error) {
 		ServerName:         cfg.ServerName,
 	}
 
-	// read ca certificates
+	// If Reader interface not provided, default to reading from File
+	if cfg.Reader == nil {
+		cfg.Reader = &fileReader{}
```

Comment:
I suggest to not overwrite `cfg.Reader`. You can just store the reader in a local variable.